source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -736,8 +736,17 @@ inline void ComponentInterface::add_input(
     RCLCPP_DEBUG_STREAM(
         this->node_logging_->get_logger(),
         "Adding input '" << parsed_signal_name << "' with topic name '" << topic_name << "'.");
-    auto subscription =
-        rclcpp::create_subscription<MsgT>(this->node_parameters_, this->node_topics_, topic_name, this->qos_, callback);
+    auto subscription = rclcpp::create_subscription<MsgT>(
```

Comment:
This is not possible unfortunately since we would have to pass the node down to the subscription handler, otherwise it's impossible to create the ROS subscription there

Reply:
I don't think there is a way to do this from the subscription handler if we don't change the interface there and then we have breaking changes. I feel that even if this wrapping is not the same as `handle_callback_exceptions` in the subscription handler, it follows the same principle: we handle CoreExceptions for the cases where we throw them internally and any exception for user defined callbacks. Meaning that here, we just catch all exceptions since we don't really expect a CoreException to be thrown from a fully user defined subscription callback

Reply:
yes, fair enough. I would still imagine that we have a slightly different structure in SubscriptionHandler/interface, but that's now going into a different direction that is out of scope. This then works for me.